src/index.js

```diff
@@ -207,28 +207,23 @@ export default function undoable (reducer, rawConfig = {}) {
     clearHistoryType: rawConfig.clearHistoryType || ActionTypes.CLEAR_HISTORY
   }
 
-  return (state, action = {}) => {
+  return (state = config.history, action = {}) => {
     debugStart(action, state)
 
-    let history
+    let history = state
     if (!config.history) {
-      debug('create history on init')
+      debug('history is uninitialized')
 
       if (state === undefined) {
-        config.history = createHistory(reducer(state, {}))
+        history = createHistory(reducer(state, {}))
+        debug('do not initialize on probe actions')
       } else if (isHistory(state)) {
-        config.history = state
+        history = config.history = state
+        debug('initialHistory properly shaped and initialized', config.history)
       } else {
-        config.history = createHistory(state)
+        history = config.history = createHistory(state)
+        debug('initialHistory reshaped and initialized', config.history)
       }
-
-      history = config.history
-    } else if (state === undefined) {
-      // If reducer is called with undefined, use our saved history as the state
-      // since that was the result of calling the reducer the initially
-      history = config.history
-    } else {
```

Comment:
If you call `reducer(undefined, { type: ANY_ACTION })` on an `undoable`, and the `config.history` is already initialized; then it will be handled by the default argument to `state`. If the `config.history` is not initialized, then it will return an empty history with `present` equal to the `initialState` of the internal reducer.

I like using the default argument of `state = config.history` because the `config.history` should be regarded as the `initialState` of an `undoable`; and in Redux a state of `undefined` should return the `initialState` for a reducer.